Core/src/Seeding/CompSpacePointAuxiliaries.cpp

```diff
@@ -488,4 +552,215 @@ void StrawLineFitAuxiliaries::updateStripResidual(
   }
 }
 
+void CompSpacePointAuxiliaries::updateTimeStripRes(
+    const Vector& sensorN, const Vector& sensorD, const Vector& stripPos,
+    const bool isBending, const double recordTime, const double timeOffset) {
+  const Vector& b1 = isBending ? sensorN : sensorD;
+  const Vector& b2 = isBending ? sensorD : sensorN;
+
+  auto positionInPlane = [&b1, &b2](const Vector& residVec) {
+    return b1 * residVec[bending] + b2 * residVec[nonBending];
+  };
+
+  /// Calculate the line intersection in the global frame
+  const Vector globIsect =
+      m_cfg.includeToF
+          ? m_cfg.localToGlobal * (positionInPlane(residual()) + stripPos)
+          : Vector::Zero();
+  /// To calculate the time of flight
+  const double globDist = globIsect.norm();
+  const double ToF = globDist / PhysicalConstants::c + timeOffset;
+  ACTS_VERBOSE("Global intersection point: "
+               << toString(globIsect) << " -> distance: " << globDist
+               << " -> time of flight takinig t0 into account: " << ToF);
+
+  m_residual[time] = recordTime - ToF;
+  constexpr auto timeIdx = static_cast<std::uint8_t>(FitParIndex::t0);
+
+  const double invDist = m_cfg.includeToF && globDist > s_tolerance
+                             ? PhysicalConstants::c / globDist
+                             : 0.;
+  for (const auto partial1 : m_cfg.parsToUse) {
+    if (partial1 == FitParIndex::t0) {
+      m_gradient[timeIdx] = -Vector::Unit(time);
+    }
+    /// Time component of the spatial residual needs to be updated
+    else if (m_cfg.includeToF) {
+      Vector& gradVec = m_gradient[static_cast<std::uint8_t>(partial1)];
+      gradVec[time] = -globIsect.dot(m_cfg.localToGlobal.linear() *
+                                     positionInPlane(gradVec)) *
+                      invDist;
+      ACTS_VERBOSE("Partial of the time residual  w.r.t. "
+                   << parName(partial1) << ": " << gradVec[time] << ".");
+    }
+  }
+
+  if (!m_cfg.useHessian || !m_cfg.includeToF) {
+    return;
+  }
+  for (const auto partial1 : m_cfg.parsToUse) {
+    for (const auto partial2 : m_cfg.parsToUse) {
+      if (partial2 > partial1) {
+        break;
+      }
+      // clang-format off
+      const auto param1 = static_cast<std::size_t>(partial1);
+      const auto param2 = static_cast<std::size_t>(partial2);
+      Vector& hessVec = m_hessian[vecIdxFromSymMat<s_nPars>(param1, param2)];
+      if (partial1 != FitParIndex::t0) {
+        hessVec[time] = -( globIsect.dot(m_cfg.localToGlobal.linear()*positionInPlane(hessVec))  +
+                           positionInPlane(gradient(partial1)).dot(positionInPlane(gradient(partial2)))) * invDist
+                        + m_gradient[param1][time] * m_gradient[param2][time] * invDist;
+        // clang-format on
```

Comment:
```suggestion
      const auto param1 = static_cast<std::size_t>(partial1);
      const auto param2 = static_cast<std::size_t>(partial2);
      Vector& hessVec = m_hessian[vecIdxFromSymMat<s_nPars>(param1, param2)];
      if (partial1 != FitParIndex::t0) {
        // clang-format off
        hessVec[time] = -( globIsect.dot(m_cfg.localToGlobal.linear()*positionInPlane(hessVec))  +
                           positionInPlane(gradient(partial1)).dot(positionInPlane(gradient(partial2)))) * invDist
                        + m_gradient[param1][time] * m_gradient[param2][time] * invDist;
        // clang-format on
```
would this be sufficient?